aten/src/ATen/native/native_functions.yaml

```diff
@@ -104,6 +104,22 @@
 - func: _unpack_dual(Tensor(a) dual, int level) -> (Tensor(a) primal, Tensor tangent)
   variants: function
 
+# NOTE [Two New-zero Functions]
+# The following two functions are used internally for forward AD
+# We need two functions here because we'd like a version to use for non-batched
+# forward that can accept non-strided tensors, and another variant specifically
+# when self is batched. Note that this is possible because a batching rule does
+# not need to redispatch to the same function.
+#
+# _new_zeros_with_same_meta needs to be CompositeExplicitAutograd for
+# jvp support in functorch
+- func: _new_zeros_with_same_meta(Tensor self, Tensor other) -> Tensor
```

Comment:
since we have a better design for this now (https://fb.quip.com/tpbLApzuoyZn#temp:C:AcS5e1e4ad6afb34bcbaa10ed365), should we go with it instead of this? There may be BC concerns if we land `_new_zeros_with_same_meta` and then want to delete it later.

Reply:
Maybe we should do this as a follow up later when we more clarity on what we need on the functorch side.
Not too concerned about the BC breaking issue since this is internal only.

Reply:
I'm not sure it is worth reviewing the batching rule in this PR for correctness (it is quite complicated), especially if we're planning on replacing it later. What other things do you think we should clarify on the functorch side? I'm pretty confident about the design as-is.

Reply:
Stepping back a bit, is this primarily a perf issue? If understand correctly nested vmap(vmap(jvp is already supported as-is (though it lacks a test) because it is using the batched fallback, so the main issue with it is perf.

I think this is one of the first things to tackle when we start thinking about perf more, but before that I think it might make more sense to get some measurements and wait until bigger optimizations for forward ad like zero tensors to get going.

> I'm not sure it is worth reviewing the batching rule in this PR for correctness (it is quite complicated)

Are there any particular parts of the batching rule that you are worried about that can cause issues? Maybe I can help with investigating that further. Overall I'm mostly confident with it because it covered by per-op testing that we do later in the stack.